### PR TITLE
Virtual thread methods

### PR DESCRIPTION
**Pull request by bhageneder**
tested on robots and changed to source code does not affect main program functionality.

### CODE REVIEW

File: src/interfaces/BaseDetectorClass.py [hunk 1]
```diff
@@ -1,6 +1,9 @@
+import queue
+
 class BaseDetector:
     def __init__(self):
-        pass
+        self.visibleQ = queue.Queue()
+        self.lostQ = queue.Queue()
```
Comment (Mitchell-Conrad-20) [inline src/interfaces/BaseDetectorClass.py:6]: good

File: src/sim/controller/classes/vRobotLink.py
Comment (Mitchell-Conrad-20) [inline src/sim/controller/classes/vRobotLink.py:1]: We shouldnt need a virtual robot link

File: src/sim/controller/classes/vRobotClass.py
Comment (Mitchell-Conrad-20) [inline src/sim/controller/classes/vRobotClass.py:1]: we shouldn't need a virtual robot class

File: src/sim/controller/config/v_global_vars.py [hunk 2]
```diff
@@ -1,8 +1,51 @@
+import threading
+import queue
+import sim.sim_global_vars as sg
+
 '''
-Contains global variables that will be used across all sim files
+Contains virtual global variables that will be used across sim files
+Each robot maintains their own virtual globals state
 '''
+class VirtualGlobals():
```
Comment (Mitchell-Conrad-20) [inline src/sim/controller/config/v_global_vars.py:9]: vGlobals

File: src/sim/controller/config/v_global_vars.py
Comment (Mitchell-Conrad-20) [inline src/sim/controller/config/v_global_vars.py:1]: good

File: src/sim/controller/functions/v_ping.py [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Send ICMP Ping out of Single Transceiver Directed at a Single Robot
+# Used for Associate (In Node Discovery) and Reassociate (in New Visible)
+def v_associate(robotIP, vg):
```
Comment (Mitchell-Conrad-20) [inline src/sim/controller/functions/v_ping.py:3]: lets be consistent with naming. pick a way. I like 'v_' for threads like you have here

File: src/sim/model/ModelClass.py [hunk 4]
```diff
@@ -22,10 +20,10 @@ class RobotModel:
...
-        self.detections = list()
         self.thread = thread
         
 
 class BlockerModel:
     def __init__(self, item):
-        self.blockerItem = item
\ No newline at end of file
+        self.blockerItem = item
+
```
Comment (Mitchell-Conrad-20) [inline src/sim/model/ModelClass.py:29]: why

File: src/sim/controller/v_main.py [hunk 5]
```diff
@@ -1,18 +1,52 @@
...
+    vg.new_visible_thread.start()
+    vg.new_lost_thread.start()
+
+    # Store Thread Number
+    thread_number = 0
+
+    while robotModel.robotItem.isActive():
+        # Blocking Call to Get New Robot
+        try:
+            robot = vg.newRobotQ.get(timeout=0.5)
```
Comment (Mitchell-Conrad-20) [inline src/sim/controller/v_main.py:38]: why are you doing timeout on a queue, its supposed to block

File: src/sim/controller/threads/v_node_discovery.py
Comment (Mitchell-Conrad-20) [inline src/sim/controller/threads/v_node_discovery.py:1]: good

**Review by Mitchell-Conrad-20 — APPROVED** [review]
I'm cool with it, would like to discuss the comments first and wait till after the milestone

File: src/sim/model/ModelClass.py (hunk 4 above)
Comment (bhageneder) [reply src/sim/model/ModelClass.py:29]: Python requires an additional line at the end of its files

File: src/sim/controller/classes/vRobotClass.py
Comment (bhageneder) [reply src/sim/controller/classes/vRobotClass.py:1]: vRobot class does not need transceiver

File: src/sim/controller/classes/vRobotLink.py
Comment (bhageneder) [reply src/sim/controller/classes/vRobotLink.py:1]: vRobotLink does not need ports, socket, or packetTime.

File: src/sim/controller/config/v_global_vars.py (hunk 2 above)
Comment (bhageneder) [reply src/sim/controller/config/v_global_vars.py]: Resolved

**Pull request merged**